File: src/lib/types/module.ts
export interface Module {
	id: string;
	program_id: string;
	title: string;
	description: string;
	order_index: number;
	color: string;
	created_at: string;
	updated_at: string;
}

export interface CreateModuleInput {
	program_id: string;
	title: string;
	description: string;
	color: string;
}

export interface UpdateModuleInput {
	title?: string;
	description?: string;
	color?: string;
}

File: src-tauri/src/db/models/module.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Module {
    pub id: String,
    pub program_id: String,
    pub title: String,
    pub description: String,
    pub order_index: i32,
    pub color: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateModuleInput {
    pub program_id: String,
    pub title: String,
    pub description: String,
    pub color: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateModuleInput {
    pub title: Option<String>,
    pub description: Option<String>,
    pub color: Option<String>,
}
